### PR TITLE
feat(arc-docs):Complete Auth Module Documentation

### LINKED ISSUE #26: Documention of Boilerplate
Documentation related to how can we use backend and other things in boilerplate

**Describe the solution you'd like**

**Describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by Deepika516**
## Description
Complete Auth Module Documentation

Fixes #26 

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Intermediate change (work in progress)

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

## Checklist:

- [ ] Performed a self-review of my own code
- [ ] npm test passes on your machine
- [ ] New tests added or existing tests modified to cover all changes
- [ ] Code conforms with the style guide
- [ ] API Documentation in code was updated
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: projects/arc-docs/src/app/docs/auth-doc/components/configure-token-doc/configure-token-doc.component.html [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+            Let's assume that your backend returns a JWT token so that we can
+            use the token payload to extract user info out of it. Each Strategy
+            specifies which token class is going to be used by default.
+          </li>
+          <li>
+            For example, NbPasswordAuthStrategy uses AuthToken, and AuthProvider
+            . It is also possible to specify another token class if it is
+            required.
+          </li>
+        </ul>
```
Comment (yeshamavani) [inline projects/arc-docs/src/app/docs/auth-doc/components/configure-token-doc/configure-token-doc.component.html:28]: explain in detail where is this to be passed
and a snippet of that

File: projects/arc-docs/src/app/docs/auth-doc/components/configure-token-doc/configure-token-doc.component.html [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+        <h2 class="nb-card-header-title">Setting a User token</h2>
+      </nb-card-header>
+      <nb-card-body>
+        <p>
+          When we get token we call authorise method, The authorize method is
+          part of an authentication service. It is designed to handle the
+          authorization process using a provided secret (such as an
+          authorization code). This method initiates a command to obtain tokens,
+          processes the response, and navigates the user based on the
+          authorization result.
```
Comment (yeshamavani) [inline projects/arc-docs/src/app/docs/auth-doc/components/configure-token-doc/configure-token-doc.component.html:43]: not making sense and this is not what we do

**Review by yeshamavani — CHANGES_REQUESTED** [review]

File: projects/arc-docs/src/app/docs/auth-doc/components/configure-token-doc/configure-token-doc.component.html [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+          Receiving user token after login/registration
+        </h2>
+      </nb-card-header>
+      <nb-card-body>
+        <ul>
+          <li>
+            At this step,It's time to get a user token but primarly we need a
+            code with the help of code we call getToken API which returns a
+            token, after a successful authentication to be able to communicate
+            with the server and, for instance, show username in the header of
```
[the remark this answers is not in the document]
Comment (Deepika516) [reply projects/arc-docs/src/app/docs/auth-doc/components/configure-token-doc/configure-token-doc.component.html:15]: Done

File: projects/arc-docs/src/app/docs/auth-doc/components/configure-token-doc/configure-token-doc.component.html (hunk 2 above)
Comment (Deepika516) [reply projects/arc-docs/src/app/docs/auth-doc/components/configure-token-doc/configure-token-doc.component.html:43]: ok

File: projects/arc-docs/src/app/docs/auth-doc/components/configure-token-doc/configure-token-doc.component.html [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+        </h2>
+      </nb-card-header>
+      <nb-card-body>
+        <ul>
+          <li>
+            At this step, it's time to get a user token. Primarily, we need a
+            code. With the help of this code, we call the getToken API, which
+            returns a token after a successful OAuth 2.0 authentication. This
+            token will allow us to communicate with the server and, for
+            instance, display the username in the header of the application.
```
Comment (yeshamavani) [inline projects/arc-docs/src/app/docs/auth-doc/components/configure-token-doc/configure-token-doc.component.html:16]:
```suggestion
            At this step, we need to authenticate the user and get users JWT token. We follow OAuth 2.0 authentication mechanism. Here we first request for code and using that code we generate the required token. This token will be used for further requests as Authorization header.
```

File: projects/arc-docs/src/app/docs/auth-doc/components/configure-token-doc/configure-token-doc.component.html [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+            This authentication can be managed using different strategies. One
+            such strategy is NbPasswordAuthStrategy. This strategy typically
+            involves the use of an AuthToken and an AuthProvider to handle
+            authentication tasks.
+            <ul>
+              <li>
+                NbPasswordAuthStrategy is a strategy provided by Nebular for
+                handling password-based authentication. It simplifies the
+                process of logging in and managing user sessions.
+              </li>
```
Comment (yeshamavani) [inline projects/arc-docs/src/app/docs/auth-doc/components/configure-token-doc/configure-token-doc.component.html:33]: give the nebular strategy link at the end to help users understand this better

File: projects/arc-docs/src/app/docs/auth-doc/components/ui-configure-doc/ui-configure-doc.component.html [hunk 6]
```diff
@@ -0,0 +1,103 @@
...
+
+    <nb-card>
+      <nb-card-header fixed>
+        <h4 class="nb-card-header-title">UI Settings</h4>
+      </nb-card-header>
+      <nb-card-body>
+        <p>
+          Alongside with the strategies' configuration AuthModule also accepts a
+          list of settings for UI side under the forms key:
+        </p>
```
Comment (yeshamavani) [inline projects/arc-docs/src/app/docs/auth-doc/components/ui-configure-doc/ui-configure-doc.component.html:93]: what is this 
can you explain this in detail

**Review by yeshamavani — CHANGES_REQUESTED** [review]

File: projects/arc-docs/src/app/docs/auth-doc/components/configure-token-doc/configure-token-doc.component.html (hunk 4 above)
Comment (Deepika516) [reply projects/arc-docs/src/app/docs/auth-doc/components/configure-token-doc/configure-token-doc.component.html:16]: sorry ? i don't understand what changes you want in this

File: projects/arc-docs/src/app/docs/auth-doc/components/ui-configure-doc/ui-configure-doc.component.html (hunk 6 above)
Comment (Deepika516) [reply projects/arc-docs/src/app/docs/auth-doc/components/ui-configure-doc/ui-configure-doc.component.html:93]: i gave the scnshort for better understanding

File: projects/arc-docs/src/app/docs/auth-doc/components/configure-token-doc/configure-token-doc.component.html (hunk 4 above)
Comment (yeshamavani) [reply projects/arc-docs/src/app/docs/auth-doc/components/configure-token-doc/configure-token-doc.component.html:16]: Changes are suggested

File: projects/arc-docs/src/app/docs/auth-doc/components/ui-configure-doc/ui-configure-doc.component.html (hunk 6 above)
Comment (yeshamavani) [reply projects/arc-docs/src/app/docs/auth-doc/components/ui-configure-doc/ui-configure-doc.component.html:93]: but I dont understand the meaning of this 
reword it

**Review by yeshamavani — CHANGES_REQUESTED** [review]

File: projects/arc-docs/src/app/docs/auth-doc/components/ui-configure-doc/ui-configure-doc.component.html (hunk 6 above)
Comment (Deepika516) [reply projects/arc-docs/src/app/docs/auth-doc/components/ui-configure-doc/ui-configure-doc.component.html]: done

File: projects/arc-docs/src/app/docs/auth-doc/components/configure-token-doc/configure-token-doc.component.html (hunk 4 above)
Comment (Deepika516) [reply projects/arc-docs/src/app/docs/auth-doc/components/configure-token-doc/configure-token-doc.component.html]: ok done

**Review by yeshamavani — APPROVED** [review]

**Pull request merged**